packages/next/src/client/components/react-dev-overlay/internal/helpers/use-error-handler.ts

```diff
@@ -60,48 +60,46 @@ export function useErrorHandler(
   }, [handleOnUnhandledError, handleOnUnhandledRejection])
 }
 
+function onUnhandledError(event: WindowEventMap['error']): void | boolean {
+  if (isNextRouterError(event.error)) {
+    event.preventDefault()
+    return false
+  }
+  handleClientError(event.error, [])
+}
+
+function onUnhandledRejection(ev: WindowEventMap['unhandledrejection']): void {
+  const reason = ev?.reason
+  if (isNextRouterError(reason)) {
+    ev.preventDefault()
+    return
+  }
+
+  if (
+    !reason ||
+    !(reason instanceof Error) ||
+    typeof reason.stack !== 'string'
+  ) {
+    // A non-error was thrown, we don't have anything to show. :-(
+    return
+  }
```

Comment:
```suggestion
  const reason: unknown = ev?.reason
  if (isNextRouterError(reason)) {
    ev.preventDefault()
    return
  }

  if (
    !isError(reason)
  ) {
    // A non-error was thrown, we don't have anything to show. :-(
    return
  }
```
to have better type coverage